File: code/dart/example/models/todo_failure.dart
import '../../lib/service_client.dart';

/// Error de dominio para operaciones sobre ToDo.
class ToDoFailure extends ServiceFailure {
  const ToDoFailure({
    required super.statusCode,
    required super.message,
    super.responseBody,
  });

  @override
  String toString() => 'ToDoFailure: $message (statusCode: $statusCode)';
}

File: code/ts/example/models/todo_failure.ts
import { ServiceFailure } from '../../src';

/// Error de dominio para operaciones sobre ToDo.
export class ToDoFailure extends ServiceFailure {
  constructor(opts: {
    statusCode: number;
    message: string;
    responseBody?: Record<string, unknown>;
  }) {
    super(opts);
  }

  override toString(): string {
    return `ToDoFailure: ${this.message} (statusCode: ${this.statusCode})`;
  }
}
